src/Driver/Http2Driver.php

```diff
@@ -743,6 +730,41 @@ private function parser(string $settings = null): \Generator
 
                 $buffer = \substr($buffer, 9);
 
+                if (++$frameCount >= $maxFramesPerSecond) {
+                    $now = $this->timeReference->getCurrentTime();
+                    if ($lastReset === $now && $bytesReceived / $frameCount < $minAverageFrameSize) {
+                        throw new Http2ConnectionException(
+                            $this->client,
+                            "Average frame size too low",
+                            self::ENHANCE_YOUR_CALM
+                        );
+                    }
+
+                    $lastReset = $now;
+                    $frameCount = 0;
+                    $bytesReceived = 0;
+                }
+
+                switch ($type) {
+                    case self::DATA:
+                    case self::HEADERS:
+                    case self::CONTINUATION:
+                        $controlFrameCount = 0;
```

Comment:
I don't think these should reset the count to `0`. Maybe take some inspiration from [nginx](https://github.com/nginx/nginx/commit/af0e284b967d0ecff1abcdce6558ed4635e3e757)?

Reply:
I prefer that approach as well. I'll put together another PR.